imports/client/components/ChatPeople.tsx

```diff
@@ -324,19 +313,20 @@ const ChatPeople = ({
         deafened: !prevState.deafened,
       };
 
-      const prevEffectiveMuteState = prevState.muted || prevState.deafened;
-      const nextEffectiveMuteState = prevState.muted || nextState.deafened;
-      // Update gain if needed
-      if (prevEffectiveMuteState !== nextEffectiveMuteState) {
-        updateGain(nextEffectiveMuteState);
-      }
+      // Pausing or unpausing the consumer in mediasoup should be sufficient to
+      // stop transmitting audio, but just to be sure, we disable the capture
+      // stream here as well.
+      audioState?.mediaSource?.getTracks().forEach((t) => {
+        // eslint-disable-next-line no-param-reassign
+        t.enabled = !nextState.muted;
+      });
 
       // Tell the server about our new state
       updatePeerState(nextState.muted, nextState.deafened);
 
       return nextState;
     });
-  }, [updateGain, updatePeerState]);
+  }, [updatePeerState]);
```

Comment:
Please fix warning about missing dependency (which I think should be fine to be just `audioState.mediaSource`, without the safe navigation).